csharp/private/macros/setup_basic_nuget_package.bzl

```diff
@@ -0,0 +1,94 @@
+load("//csharp/private:providers.bzl", "CSharpAssembly")
+load("//csharp/private:rules/imports.bzl", "import_library", "import_multiframework_library")
+
+def _import_dll(dll, has_pdb, imports):
+    path = dll.split("/")
+
+    tfm = path[1]
+
+    # Ignore frameworks we don't support (like net35)
+    if tfm not in CSharpAssembly:
+        return
+
+    lib_name = path[-1].rsplit(".", 1)[0]
+
+    target_name = "%s-%s" % (lib_name, tfm)
+
+    if lib_name not in imports:
+        imports[lib_name] = {tfm: target_name}
+    else:
+        imports[lib_name][tfm] = target_name
+
+    if dll in has_pdb:
+        import_library(
+            name = target_name,
+            target_framework = tfm,
+            dll = dll,
+            pdb = dll[:-3] + "pdb",
+        )
+    else:
+        import_library(
+            name = "%s-%s" % (lib_name, tfm),
+            target_framework = tfm,
+            dll = dll,
+        )
+
+def setup_basic_nuget_package():
+    """This macro gets used to implement the default NuGet BUILD file.
+
+       We are limited by the fact that Bazel does not allow the analysis phase to
+       read the contents of source files, e.g. to correctly configure deps. For
+       more advanced usages a BUILD file will need to be generated outside of
+       Bazel.
+
+       This has to be public so that packages can call it but you probably
+       shouldn't use it directly.
+    """
+    dlls = native.glob(["lib/*/*.dll"])
```

Comment:
For nuget packages, how are envisioning the handling of [runtime + other](https://github.com/dotnet/corefx/tree/master/pkg/Microsoft.NETCore.Platforms#rid-qualified-assets-are-preferred)  assets?

Will this be including some sort of priority analysis similar to how [FrameworkCompatibility ](https://github.com/Brightspace/rules_csharp/blob/master/csharp/private/providers.bzl#L88) exists?

Reply:
For runtimes my gut feel is that we'd just change which DLL we put in place for the `dll` attr of `import_library`. That might be tricky because I'm not sure if the loading phase knows about your target platform yet. Alternatively I think we'd only need to make this a feature of `import_library` but I'm not confident.

For now I don't think we need to support it but it would be good to error out if that folder exists. None of the packages that D2L uses in the mono-repo have runtime-specific variants.